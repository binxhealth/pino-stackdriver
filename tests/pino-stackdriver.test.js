const { join } = require('path')
const spawn = require('child_process').spawn

const pinoStackdriver = join(__dirname, '../')
const lineOne = JSON.stringify({
  hostname: 'host',
  level: 50,
  msg: 'TestError: blah blah blah',
  pid: 1,
  req: { path: '/test' },
  res: { statusCode: 500 },
  responseTime: 116,
  v: 1
})
const lineTwo = JSON.stringify({
  hostname: 'host',
  level: 30,
  msg: 'A user forgot their password',
  pid: 1,
  req: { path: '/api/password-reset' },
  res: { statusCode: 201 },
  responseTime: 288,
  v: 1
})

test('pino-stackdriver adds severity to log entry', done => {
  const child = spawn('node', [pinoStackdriver])
  child.on('error', done.fail)
  child.stdout.on('data', data => {
    console.log('data', data.toString())
  })
  child.stdin.write(lineOne)
  child.stdin.write(lineTwo)
  child.kill()
  done()
})
