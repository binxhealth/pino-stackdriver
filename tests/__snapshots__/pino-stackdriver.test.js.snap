// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pino-stackdriver adds severity to log entry 1`] = `
Object {
  "hostname": "host",
  "level": 50,
  "msg": "TestError: blah blah blah",
  "pid": 1,
  "req": Object {
    "path": "/test",
  },
  "res": Object {
    "statusCode": 500,
  },
  "responseTime": 116,
  "severity": "ERROR",
  "v": 1,
}
`;

exports[`pino-stackdriver adds severity to log entry 2`] = `
Object {
  "hostname": "host",
  "level": 30,
  "msg": "A user forgot their password",
  "pid": 1,
  "req": Object {
    "path": "/api/password-reset",
  },
  "res": Object {
    "statusCode": 201,
  },
  "responseTime": 288,
  "severity": "INFO",
  "v": 1,
}
`;
